File: Gemini_Pro/Python/Factory_Method/Round1/CafeTest1.py
# CafeTest.py
import pytest
from Cafe1 import BeverageFactory, PastryFactory, SnackFactory, order_item  # นำเข้าฟังก์ชันและคลาสที่ต้องการทดสอบจากไฟล์ Cafe.py

# Test Cases
def test_order_espresso():
    factory = BeverageFactory()
    assert order_item(factory, "Espresso") == "Preparing Espresso"

def test_order_latte():
    factory = BeverageFactory()
    assert order_item(factory, "Latte") == "Preparing Latte"

def test_order_milkshakes():
    factory = BeverageFactory()
    assert order_item(factory, "Milkshakes") == "Preparing Milkshakes"

def test_order_fruit_juices():
    factory = BeverageFactory()
    assert order_item(factory, "Fruit Juices") == "Preparing Fruit Juices"

def test_order_croissant():
    factory = PastryFactory()
    assert order_item(factory, "Croissant") == "Preparing Croissant"

def test_order_chocolate_cake():
    factory = PastryFactory()
    assert order_item(factory, "Chocolate Cake") == "Preparing Chocolate Cake"

def test_order_chocolate_chip_cookies():
    factory = PastryFactory()
    assert order_item(factory, "Chocolate Chip Cookies") == "Preparing Chocolate Chip Cookies"

def test_order_tuna_sandwich():
    factory = SnackFactory()
    assert order_item(factory, "Tuna Sandwich") == "Preparing Tuna Sandwich"

def test_order_spaghetti():
    factory = SnackFactory()
    assert order_item(factory, "Spaghetti") == "Preparing Spaghetti"

def test_order_vegetable_salad():
    factory = SnackFactory()
    assert order_item(factory, "Vegetable Salad") == "Preparing Vegetable Salad"

def test_order_invalid_item():
    factory = BeverageFactory()
    assert order_item(factory, "Invalid Item") == "Item not available"

File: Gemini_Pro/Java/Factory_Method/Round1/CafeTest1.java
import org.junit.jupiter.api.Test;

import code.Item;

import static org.junit.jupiter.api.Assertions.*;

public class CafeTest1 {

    @Test
    void testGetEspresso() {
        Item item = Cafe.getItem("Beverage", "Espresso");
        assertEquals("Espresso", item.getName());
        assertEquals(3.50, item.getPrice());
    }

    @Test
    void testGetLatte() {
        Item item = Cafe.getItem("Beverage", "Latte");
        assertEquals("Latte", item.getName());
        assertEquals(3.50, item.getPrice());
    }

    @Test
    void testGetMilkshakes() {
        Item item = Cafe.getItem("Beverage", "Milkshakes");
        assertEquals("Milkshakes", item.getName());
        assertEquals(3.50, item.getPrice());
    }

    @Test
    void testGetFruitJuices() {
        Item item = Cafe.getItem("Beverage", "Fruit Juices");
        assertEquals("Fruit Juices", item.getName());
        assertEquals(3.50, item.getPrice());
    }

    @Test
    void testGetCroissant() {
        Item item = Cafe.getItem("Pastry", "Croissant");
        assertEquals("Croissant", item.getName());
        assertEquals(2.00, item.getPrice());
    }

    @Test
    void testGetChocolateCake() {
        Item item = Cafe.getItem("Pastry", "Chocolate Cake");
        assertEquals("Chocolate Cake", item.getName());
        assertEquals(2.00, item.getPrice());
    }

    @Test
    void testGetChocolateChipCookies() {
        Item item = Cafe.getItem("Pastry", "Chocolate Chip Cookies");
        assertEquals("Chocolate Chip Cookies", item.getName());
        assertEquals(2.00, item.getPrice());
    }

    @Test
    void testGetTunaSandwich() {
        Item item = Cafe.getItem("Snack", "Tuna Sandwich");
        assertEquals("Tuna Sandwich", item.getName());
        assertEquals(4.00, item.getPrice());
    }

    @Test
    void testGetSpaghetti() {
        Item item = Cafe.getItem("Snack", "Spaghetti");
        assertEquals("Spaghetti", item.getName());
        assertEquals(4.00, item.getPrice());
    }

    @Test
    void testGetVegetableSalad() {
        Item item = Cafe.getItem("Snack", "Vegetable Salad");
        assertEquals("Vegetable Salad", item.getName());
        assertEquals(4.00, item.getPrice());
    }

    @Test
    void testGetInvalidItem() {
        Item item = Cafe.getItem("Invalid", "Invalid");
        assertNull(item);
    }
}